src/onzr/tui.py

```diff
@@ -0,0 +1,133 @@
+"""Onzr: TUI elements."""
+
+from textual.app import App, ComposeResult
+from textual.containers import Container, HorizontalGroup, Vertical
+from textual.reactive import reactive
+from textual.widgets import (
+    Button,
+    DataTable,
+    Footer,
+    Header,
+    Static,
+    TabbedContent,
+    TabPane,
+)
+
+from onzr import cli
+from onzr.client import OnzrClient
+from onzr.models import PlayingState, QueuedTracks
+
+
+def get_now_playing():
+    """Get the current playing status."""
+    client = OnzrClient()
+    playing_state: PlayingState = client.now_playing()
+    if playing_state.track is not None:
+        return playing_state.track.title
+    else:
+        return "Nothing"
+
+
+class PlayStatusWidget(Static):
+    """Show the current playing status."""
+
+    now_playing_text = reactive("Nothing", layout=True)
+
+    def render(self) -> str:
+        """Render the current playing status."""
+        return f"Now playing: {self.now_playing_text}!"
+
+
+class PlayControl(HorizontalGroup):
+    """A group of buttons to control play/pause/stop music."""
+
+    now_playing_text = reactive("Nothing", layout=True)
+
+    def on_button_pressed(self, event: Button.Pressed) -> None:
+        """Event handler called when a button is pressed."""
+        button_id = event.button.id
+
+        match button_id:
+            case "play":
+                cli.play()
+            case "pause":
+                cli.pause()
+            case "stop":
+                cli.stop()
+            case "previous":
+                cli.previous()
+            case "next":
+                cli.next()
+
+    def compose(self) -> ComposeResult:
+        """Create child widgets of a stopwatch."""
+        yield Button("Play", id="play", variant="primary", classes="play-button")
+        yield Button("Pause", id="pause", variant="primary", classes="play-button")
+        yield Button("Stop", id="stop", variant="primary", classes="play-button")
+        yield Button(
+            "Previous", id="previous", variant="primary", classes="play-button"
+        )
+        yield Button("Next", id="next", variant="primary", classes="play-button")
+
+
+class OnzrTuiApp(App):
+    """OnzrTui app.
+
+    This is the class that will:
+    - define the TUI, compose the interface with the widgets
+    - gather the information about the playlist
+    - allow features such as search...
+    """
+
+    CSS_PATH = "tui.tcss"
+
+    BINDINGS = [
+        ("p", "show_tab('playlist-tab')", "Play list"),
+        ("s", "show_tab('search-tab')", "Search"),
+        ("ctrl+q", "quit", "Quit"),
+    ]
+
+    def on_mount(self) -> None:
+        """Initialize the app."""
+        self.title = "ONZR: play your music from the terminal"
+        self.set_interval(1, self.update_status)
```

Comment:
Does the following work to add the API client to the state?


```suggestion
        self.onzr_client = get_onzr_client()
        self.set_interval(1, self.update_status)
```